richtext-ui/src/main/java/com/zachklipp/richtext/ui/Heading.kt

```diff
@@ -92,7 +92,19 @@ internal val DefaultHeadingStyle: HeadingStyle = { level, textStyle ->
 
   val richTextStyle = currentRichTextStyle.resolveDefaults()
   val headingStyleFunction = richTextStyle.headingStyle!!
-  val currentTextStyle = resolveDefaults(AmbientTextStyle.current, LayoutDirectionAmbient.current)
+
+  // There is an unintended bug regarding Heading color in Dark Mode.
+  // It is because `resolveDefaults` function overrides the Unspecified text color without
+  // considering current theme. Unless `resolveDefaults` is changed to a composable function,
+  // this bug will stay in here. So, we need to specify the text color before passing it to
+  // `resolveDefault` function as a parameter.
+  val currentTextStyle = resolveDefaults(
+    AmbientTextStyle.current.merge(TextStyle(
+      color = AmbientContentColor.current
+    )),
+    LayoutDirectionAmbient.current
+  )
```

Comment:
This is a lot of nesting in this parameter, I think it would be more readable if an intermediate variable were used. E.g.
```suggestion
  val currentColorStyle = TextStyle(color = AmbientContentColor.current)
  val currentTextStyle = resolveDefaults(
    AmbientTextStyle.current.merge(currentColorStyle),
    LayoutDirectionAmbient.current
  )
```

Reply:
Or maybe this should be more explicit:
```suggestion
  val incomingStyle = AmbientTextStyle.current.let {
    if (it.textColor == Unspecified) {
      it.copy(textColor = AmbientContentColor.current
    } else {
      it
    }
  }
  val currentTextStyle = resolveDefaults(incomingStyle, LayoutDirectionAmbient.current)
```

I like the explicitness because it makes it really clear exactly what problem we're fixing.